### PR TITLE
fix(cf-harness): require decisions on retained pattern reuse

### PR DESCRIPTION
**Pull request by bfollington**
Research selected the reusable mailbox component in rehearsal b544047f and handed it to the pattern-author, but the author rewrote the query without importing it or explaining why. `run_pattern` now checks retained research selections before compiling new source: import each selection as `cf:pattern:<id>`, or give `reuseReasons[<id>]` as one nonblank line explaining why it does not fit this call.

The check uses the existing context selector (latest orientation and two answers), compiler import reader, and Fabric reference parser. It applies to parent and child authors, including incomplete kits and historical findings. Leads and unselected inspected records remain advisory; direct indexed execution is unchanged. Narrower atoms and readers of existing results can explain that scope in free text. Refusals list exact retained identities and both retry options, before Fabric access or compilation. Reasons stay in the tool-call record.

Shared guidance reaches research handoffs, child context, and the tool descriptor. The research schema describes selections as reusable building blocks, and the invocation schema requires each reason to be one nonblank line. Published-component imports are explicitly permitted in the descriptor.

Current head: `59c92a41e6`. Its two-file documentation follow-up adds the gate to the detailed tool contract and central map panel, with both snapshots pinned to the checkoutable implementation commit `688cd617a6`. All production and test bytes are unchanged from that commit.

Validation on the implementation tree (`688cd617a6`, base `1fb678d2ab`, including #7828 and #7832):

- Recorded b544047f task and reduced research kits exercise parent refusal and delegated-child refusal before Fabric access. Real indexed composition and an explained narrower atom execute successfully.
- cf-harness: 1,089 tests / 3,579 steps / zero failures. Root typecheck: 398 paths / 46 groups. All 21 formatting, lint, documentation, static, and diff gates passed at the rebase head. After the final status correction, the full package suite, root typecheck, formatting, lint, test-alias, conflict-marker, and diff checks passed again.
- System map and research panel rendered at 1440×1000. Both snapshots were rebuilt from current main with this change and refreshed stamps.
- Earlier mutation evidence: 26/26 assertion and 3/3 production mutants killed; the schema follow-up killed 4/4 assertion and 1/1 schema mutants. These remain attributed to those original executions; the rebase changed no guard or test logic.
- The final Cubic correction classifies a malformed-import reuse preflight refusal as `error`, reserving `compile-error` for the compiler path. Its 55-step focused suite passes; restoring the wrong status and inverting the changed assertion each fail the regression test. A test-history alias preserves the renamed case's identity.
- CI on `688cd617a6` passed on attempt 2. Runner Tests (8/8) initially hit the known `awaitAdmitted` stuck-wait guard in the failure-park backoff case; the same runner bytes had passed at `15d5717819`, and the single unchanged shard rerun passed. The documentation-only follow-up passed format, lint, documentation (612 blocks), history, conflict-marker, diff, and browser-render checks; final-head CI remains separately reported by GitHub.

Conflict resolution retains main's cell-read/shared-query ceiling guidance beside the reuse rule. A documentation correction also qualifies CURRENT_STATE's persistence statement for preflight refusals. No new runtime behavior was added during the rebase.

This establishes an import or explicit explanation. It does not prove meaningful invocation or judge the explanation's correctness. The recorded regression is deterministic; no new live-model timing or reuse-rate claim is made.

Fixes CT-2375.

### CODE REVIEW

**Review by bfollington — COMMENTED** [review]
cf-review via Codex (GPT-6), on behalf of @bfollington.

Scope: CT-2375 retained-pattern reuse, 15 files in cf-harness. Deep-read the production and test delta, context selection/child inheritance, Fabric import parsing, source-attempt collapse, and the documentation ripple.

No blocking findings. One guard and one shared instruction enforce an import or a free-text one-line reason before new source accesses Fabric. The existing compiler import collector and reference parser distinguish dependencies from comments/string examples. Exact IDs remain qualified in refusal text so the ordinary model scrub preserves them. The bounded context selector is reused; unverified leads and direct indexed calls are unchanged. Reason fields remain in the tool-call record when source attempts collapse.

The b544047f fixture covers the actual incomplete retained selections and task through both parent and child paths. Real composition and explanation paths execute; the child refusal test proves no Fabric session is opened. Mutation tests kill the removed guard, invalid-reason acceptance, and false import detection. All 26 changed assertion matchers also fail when inverted.

The rebase resolved two documentation stamps only and retained #7819's publication identity. The guard intentionally establishes a dependency or explanation rather than semantic adequacy; there is no new live-model performance claim.

Verdict: ready subject to CI and current-head automated review. Ben owns merge.

File: packages/cf-harness/src/contracts/run-pattern.ts [hunk 1]
```diff
@@ -41,6 +44,12 @@ export const RUN_PATTERN_INPUT_SCHEMA = {
       description:
         "Id of a pattern published to the index, as search_patterns reports it. Exactly one of sourceText and patternId is given; the published program is fetched and compiled without passing through this conversation.",
     },
+    reuseReasons: {
+      type: "object",
+      additionalProperties: { type: "string", minLength: 1 },
```
[the remark this answers is not in the document]
Comment (bfollington) [reply packages/cf-harness/src/contracts/run-pattern.ts]: Fixed in 74ee1cecf1. The input schema now requires a nonblank string with no CR or LF, including a trailing newline. Tests exercise the advertised pattern and the existing runtime predicate with padded valid text, whitespace-only text, embedded line breaks, and trailing LF/CRLF. Weakening the schema to accept any string kills the regression; all four assertions in the changed cases also fail when inverted. No new validator or helper.

File: packages/cf-harness/README.md [hunk 2]
```diff
@@ -1689,6 +1689,19 @@ bindings, and historical bindings transfer nothing. The parent's full CFC
 context carries forward even when selection omits a result. Pattern authors
 start from these findings and ask only unresolved questions.
 
+Before compiling new `run_pattern.sourceText`, the host checks selections from
```
[the remark this answers is not in the document]
Comment (bfollington) [reply packages/cf-harness/README.md:1735]: Fixed in 74ee1cecf1. The README now states that a retained-pattern preflight refusal returns before opening Fabric or compiling and persists nothing. It retains the no-deletion and source-history retention-root semantics for pieces actually created.

**Review by bfollington — COMMENTED** [review]
cf-review via Codex (GPT-6), on behalf of @bfollington.

Review follow-up at 74ee1cecf1: two P2 findings addressed in three files (+33/-6). Deep-read the schema pattern, edge cases, and persistence wording; no blocking finding. Focused tests passed (17 steps), the complete harness suite passed (1,081 tests / 3,499 steps), repository typecheck passed all 46 groups, and root format, lint and documentation checks passed. Four assertion inversions and the permissive-schema mutant all failed as required. CI and Cubic on this head remain pending.

**Review by bfollington — COMMENTED** [review]
cf-review via Codex (GPT-6), on behalf of @bfollington.

Scope: retained-pattern reuse decisions at head `15d57178191e440d95597942405e873260944c23` against main `1fb678d2abf93593f41d467cb85cd8d7b7e73d9f`. Deep-read the changed production and test code, its context selector and invocation boundary, the documentation changes, and the rebase range-diff.

No blocking findings. The preflight uses the existing retained-context selector, compiler import reader, and Fabric reference parser. It accepts an actual component import or an explicit one-line explanation before Fabric access; comments, quoted examples, unverified leads, and direct indexed execution retain their documented behavior. The refusal preserves qualified pattern identities and gives both retry routes. The regression exercises the recorded rehearsal selection through the parent and delegated author, and a real composition plus an explained narrower atom.

The rebase preserves main's updated cell-read and shared-query confidentiality guidance alongside this PR's component-import and reuse guidance. Guard, parser, schema, and test logic are unchanged by conflict resolution. The two snapshot files were reconstructed from current main with the reuse text and refreshed stamps; main's retraction, submit_result, and foreign-space contracts remain present. Self-review also aligned CURRENT_STATE's persistence limitation with the README: a preflight refusal creates no piece. This is documentation only.

Validation on the final tree: 1,089 cf-harness tests / 3,579 steps / zero failures; 398 checked paths / 46 groups; all 21 formatting, lint, documentation, static, and diff gates passed. The map and research panel were rendered and inspected at 1440×1000, with no page errors; the isolated browser is closed. Earlier mutation evidence remains attributed to the original implementation and schema follow-up: 26 assertion and 3 production mutants, then 4 assertion and 1 schema mutant, all killed. Those mutations were not rerun or reattributed to this rebase.

This enforces an import or explanation, not meaningful invocation or the explanation's correctness. The fixture is a recorded regression; no new live-model timing or reuse-rate claim is made.

Verdict at `15d57178191e440d95597942405e873260944c23`: sound to merge once this head's CI and automated review settle green. Ben owns merge.

**Comment by bfollington** [comment]
@cubic review

File: packages/cf-harness/src/tools/run-pattern.ts [hunk 3]
```diff
@@ -1000,6 +1004,24 @@ export const runPatternTool: HarnessToolDefinition<
...
       }
+      let unexplained;
+      try {
+        unexplained = await unexplainedResearchPatterns(
+          context.researchRuns ?? [],
+          { name: RUN_PATTERN_SOURCE_MAIN, contents: sourceText },
+          input.reuseReasons,
+        );
+      } catch (error) {
+        return errorOutput("compile-error", errorMessage(error));
```
[the remark this answers is not in the document]
Comment (bfollington) [reply packages/cf-harness/src/tools/run-pattern.ts]: Fixed in `688cd617a6cb2b1e777d7ca61b287b9a3f54a4d3`. The reuse preflight catch now returns the existing `error` status, before Fabric access or compilation. The `compileAndSavePattern` catch continues to return `compile-error`.

The malformed `cf:pattern:invalid` fixture now asserts `status: "error"`; its renamed identity is carried by the test-history alias. The focused suite passes all 55 steps. Restoring `compile-error` in production fails this exact case, as does inverting the changed assertion (2/2 mutants killed). The full package suite passes 1,089 tests / 3,579 steps, and root typecheck, formatting, lint, and alias checks pass.

**Review by bfollington — COMMENTED** [review]
cf-review via Codex (GPT-6), on behalf of @bfollington.

Current head: `688cd617a6cb2b1e777d7ca61b287b9a3f54a4d3`. Addendum to the review at `15d5717819`; deep-read the three-file review correction and the preflight/compiler error paths.

No blocking findings. The sole production change returns `error` from reuse preflight, which runs before Fabric access or compilation. Actual compiler failures keep `compile-error`, matching the author's source-repair guidance. The fixture tests a real malformed Fabric import; the test-history alias preserves the renamed test's identity. No new status, helper, or policy path.

Validation: 55 focused steps; 1,089 package tests / 3,579 steps; 398 typechecked paths / 46 groups; root formatting and lint, test-alias, conflict-marker, and diff checks all pass. Restoring the misclassification and inverting the changed assertion each kill the regression test (2/2). The full package invocation completed without a restart after a slow CLI skill-script case. Prior map and broader static checks remain attributed to `15d5717819`; those surfaces are unchanged.

Verdict: the correction is ready for CI and Cubic on this head. Ben owns the merge.

**Comment by bfollington** [comment]
@cubic review

File: packages/cf-harness/docs/CURRENT_STATE.md [hunk 4]
```diff
@@ -140,9 +140,13 @@ The current package provides:
   transfer reads only selected current bindings; its inherited CFC context
   retains the full parent influence even when no kit is selected. Kits and
   confirmed records persist through delegation, while local authored-source
-  artifacts record the research ids that shaped them. `query_docs` is accepted
-  only as a legacy CLI or persisted-policy alias and is normalized without
-  rewriting old transcript or run-state evidence;
+  artifacts record the research ids that shaped them. Before new source
+  compiles, `run_pattern` requires an import or a one-line
```
[the remark this answers is not in the document]
Comment (bfollington) [reply packages/cf-harness/docs/CURRENT_STATE.md:144]: Fixed in `59c92a41e621f0c20ade3202758dbb9f87f5d6ab`. The detailed fabric-session tool description now states the retained-selection import-or-`reuseReasons` gate, including incomplete kits, and explains that omitting both returns `error` with the retained identities and both retry paths before any piece is persisted. It distinguishes advisory leads and direct indexed execution. No production or test bytes changed; documentation checks pass.

File: packages/cf-harness/docs/system-map/cfc-system-map.html [hunk 5]
```diff
@@ -671,7 +671,7 @@ <h1>
     // map is re-verified; the stamp is drawn in the canvas's top-left corner.
     const SNAPSHOT = {
       date: "2026-09-22",
-      sha: "4e3d5274b4+console-retraction",
+      sha: "1fb678d2ab+research-reuse",
```
[the remark this answers is not in the document]
Comment (bfollington) [reply packages/cf-harness/docs/system-map/cfc-system-map.html]: Fixed in `59c92a41e621f0c20ade3202758dbb9f87f5d6ab`. Both stamps now name `688cd617a6`, the verified implementation revision, without a working-tree suffix. That commit contains the preflight behavior and can be checked out directly; this follow-up changes only the two documentation files. The revised stamp and panel were verified in the browser at 1440×1000.

File: packages/cf-harness/docs/system-map/cfc-system-map.html [hunk 6]
```diff
@@ -671,7 +671,7 @@ <h1>
     // map is re-verified; the stamp is drawn in the canvas's top-left corner.
```
[the remark this answers is not in the document]
Comment (bfollington) [reply packages/cf-harness/docs/system-map/cfc-system-map.html:671]: Fixed in `59c92a41e621f0c20ade3202758dbb9f87f5d6ab`. The central `e_run_pattern` panel now states the preflight before Fabric access or compilation, `cf:pattern:<id>` or one nonblank `reuseReasons[<id>]` line for each retained selection (including incomplete kits), and the error/no-persistence outcome when both are omitted. Advisory leads and direct indexed execution are distinguished. The panel was rendered and inspected at 1440×1000; no drawing or runtime code changed.

**Review by bfollington — COMMENTED** [review]
cf-review via Codex (GPT-6), on behalf of @bfollington.

Current head: `59c92a41e621f0c20ade3202758dbb9f87f5d6ab`. Addendum to the review at `688cd617a6`; deep-read the complete two-file documentation delta against the implemented preflight and rendering.

No blocking findings. The detailed `run_pattern` contract and its central map panel now name the import-or-one-line-`reuseReasons` gate, its incomplete-kit scope, and its refusal before Fabric access or compilation. Both stamps name the actual verified implementation commit `688cd617a6`, which can be checked out to reproduce the described behavior. All production and test bytes are unchanged from that revision.

Validation: repository formatting and lint, documentation (612 checked blocks), history-index, conflict-marker, and diff checks pass. The final map paragraph and stamp were inspected at 1440×1000; no drawing code changed. The 1,089-test / 3,579-step package result, 398-path typecheck, and mutation evidence stay attributed to `688cd617a6` and earlier recorded executions.

Verdict: the three documentation findings are resolved. CI and Cubic must settle on this head before READY; Ben owns the merge.

**Comment by bfollington** [comment]
@cubic review
